librz/asm/arch/luac/v54/disassembly_54.c

```diff
@@ -0,0 +1,356 @@
+		break;
+
+		/* iABC - A & C instructions */
+		// TODO : In and Out Status
+	case OP_TFORCALL: /*	A C	R[A+4], ... ,R[A+3+C] := R[A](R[A+1], R[A+2]);	*/
+	case OP_VARARG: /*	A C	R[A], R[A+1], ..., R[A+C-2] = vararg		*/
+		asm_string = luaop_new_str_2arg_ex(
+			opnames[opcode],
+			a, c,
+			NULL,
+			LUA_NO_PREFIX, LUA_NO_PREFIX);
+		break;
+
+		/* iABC - single A instructions */
+	case OP_LOADKX: /*	A	R[A] := K[extra arg] */
+		asm_string = luaop_new_str_1arg(opnames[opcode], a, LUA_KX_MARK);
+		break;
+
+	case OP_LOADFALSE: /*	A	R[A] := false					*/
+	case OP_LFALSESKIP: /*  A	R[A] := false; pc++				*/
+	case OP_LOADTRUE: /*	A	R[A] := true					*/
+	case OP_CLOSE: /*	A	close all upvalues >= R[A]			*/
+	case OP_TBC: /*	        A	mark variable A "to be closed"			*/
+	case OP_RETURN1: /*	A	return R[A]					*/
+	case OP_VARARGPREP: /*  A	(adjust vararg parameters)			*/
+		asm_string = luaop_new_str_1arg(opnames[opcode], a, NULL);
+		break;
+
+		/* iABC - special instructions */
+	case OP_TEST: /*	A k	if (not R[A] == k) then pc++			*/
+		asm_string = luaop_new_str_2arg_ex(
+			opnames[opcode],
+			a, isk,
+			NULL,
+			LUA_REG_PREF, LUA_NO_PREFIX);
+		break;
+
+	case OP_RETURN0: /*		return						*/
+		asm_string = rz_str_newf("RETURN0");
+		break;
+
+		/* iABx instructions */
+	case OP_LOADK: /*	A Bx	R[A] := K[Bx]					*/
+		asm_string = luaop_new_str_2arg_ex(
+			opnames[opcode],
+			a, bx,
+			NULL,
+			LUA_REG_PREF, LUA_CONST_PREF);
+		break;
+
+		// TODO : PC status
+	case OP_FORLOOP: /*	A Bx	update counters; if loop continues then pc-=Bx; */
+	case OP_FORPREP: /*	A Bx	<check values and prepare counters>;
+                     if not to run then pc+=Bx+1;			*/
+	case OP_TFORPREP: /*	A Bx	create upvalue for R[A + 3]; pc+=Bx		*/
+	case OP_TFORLOOP: /*	A Bx	if R[A+2] ~= nil then { R[A]=R[A+2]; pc -= Bx }	*/
+		asm_string = luaop_new_str_2arg_ex(
+			opnames[opcode],
+			a, bx,
+			LUA_JMP_MARK,
+			LUA_NO_PREFIX, LUA_NO_PREFIX);
+		break;
+
+	case OP_CLOSURE: /*	A Bx	R[A] := closure(KPROTO[Bx])			*/
+		asm_string = luaop_new_str_2arg_ex(
+			opnames[opcode],
+			a, bx,
+			LUA_CLOSURE_MARK,
+			LUA_REG_PREF, LUA_KPROTO_PREF);
+		break;
+
+		/* iAsBx instructions */
+	case OP_LOADI: /*	A sBx	R[A] := sBx					*/
+	case OP_LOADF: /*	A sBx	R[A] := (lua_Number)sBx				*/
+		asm_string = luaop_new_str_2arg_ex(
+			opnames[opcode],
+			a, sbx,
+			NULL,
+			LUA_REG_PREF, LUA_NO_PREFIX);
+		break;
+
+		/* iAx instructions */
+	case OP_EXTRAARG: /*	Ax	extra (larger) argument for previous opcode	*/
+		asm_string = luaop_new_str_1arg_ex(
+			opnames[opcode],
+			ax,
+			NULL,
+			LUA_NO_PREFIX);
+		break;
+
+		/* isJ instructions */
+		// TODO : PC status
+	case OP_JMP: /*	        sJ	pc += sJ					*/
+		asm_string = luaop_new_str_1arg_ex(
+			opnames[opcode],
+			sj,
+			LUA_JMP_MARK,
+			LUA_NO_PREFIX);
+		break;
+
+	default:
+		asm_string = rz_str_newf("INVALID");
+	}
+
+	rz_strbuf_append(&op->buf_asm, asm_string);
+	op->size = 4;
```

Comment:
is this right? I saw in the analysis part that some instructions had an extra argument coming from the next 4 bytes.

Reply:
Yeah, in fact the next instruction of `OP_LOADKX` and `OP_NEWTABLE` is always a `OP_EXTRAARG`, the extra argument is also treated as an instruction to keep the same instruction size. 
And in analysis routine, I didn't find a proper OP_TYPE for that. So I set the size of opcode as 8 to skip next instruction. I will check the lua source and my analysis part again , thanks for your advices ~